operator/v1/types_insights.go

```diff
@@ -0,0 +1,124 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// enabled marks the health check as enabled
+	HealthCheckEnabled HealthCheckState = "Enabled"
+	// disabled marks the health check as disabled
+	HealthCheckDisabled HealthCheckState = "Disabled"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// InsightsOperator holds cluster-wide information about the Insights Operator.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type InsightsOperator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights.
+	// +kubebuilder:validation:Required
+	Spec InsightsOperatorSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator.
+	// +optional
+	Status InsightsOperatorStatus `json:"status"`
+}
+
+type InsightsOperatorSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsOperatorStatus struct {
+	OperatorStatus `json:",inline"`
+	// gatherStatus provides basic information about the last Insights data gathering.
+	// When omitted, this means no data gathering has taken place yet.
+	// +optional
+	GatherStatus GatherStatus `json:"gatherStatus,omitempty"`
+	// reportStatus provides general Insights analysis results.
+	// +optional
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatherStatus struct {
+	// lastGatherTime is the last time when Insights data gathering finished.
```

Comment:
Do we clear this or change it at all when one is in progress? Did we work out how a user would work out that a gather is in progress? Or are we deferring that til we introduce the job style API later?

We should note what this means when it's empty or zero

Reply:
No we don't (it can happen that a gathering is interrupted - i.e will not finish [e.g pod restart] and the time will not be set). We didn't (the information is not available now anyway) and yes I would defer this problem to force gather/job design. 

So empty or zero means that there hasn't been any data gathering yet and yes let's document it here.